005-tailwindcss-framework.md

```diff
@@ -0,0 +1,126 @@
+- Support for [responsiveness](https://tailwindcss.com/docs/responsive-design) and [pseudo-classes](https://tailwindcss.com/docs/pseudo-class-variants) within the library of generated utility classes.
+- Tailwind also provides the ability to extract small components of repeating utility class combinations when needed. While Tailwind encourages always using utility classes in your HTML, there can be scenarios where you may want to extract a component that includes multiple utility classes into a class for use in various places. Since we use React and build most of our interfaces as components, this will be unlikely, but it's a nice feature that Tailwind provides to help with maintainability if some blocks of HTML are not componentized. See the [Tailwind documentation](https://tailwindcss.com/docs/extracting-components) on extracting components for more information.
+- Repeating class names in HTML is actually pretty great for GZIP, resulting in smaller HTML files with an already reduced-size CSS file of utility classes.
+
+Additional benefits:
+
+- [TailwindCSS](https://github.com/tailwindcss/tailwindcss) is open source so we could contribute back and help further promote open source software!
+- Actively maintained.
+- Available Tailwind CSS IntelliSense plugin for [VS Code](https://marketplace.visualstudio.com/items?itemName=bradlc.vscode-tailwindcss).
+
+## Drawbacks
+
+Once Tailwind CSS is implemented, it will take our full-stack developers a little bit of time to familiarize themselves with the utility class syntax. However the Tailwind CSS documentation is very thorough and easy to search, acting as a great reference. The VS Code extension is also great for suggesting class names based on a project's configuration.
+
+Depending on the complexity of a component, seeing a series of styles displayed as class names within the HTML could become a little harder to read and potentially difficult to spot typos in a class names or potential duplicates. This could be mitigated by establishing a rule whereby class names are added alphabetically or there may be an ESLint rule we could apply; something we can figure out best practices as a team.
+
+There may also be some time to adjust and learn how to best apply utility classes. For example, whether it is best to apply a font-size utility class to a parent container like a `<div>` or to apply it to each individual child `<p>` tag with in parent container. These situations will be tackled and addressed as we encounter them, establishing best practices as a team.
+
+## Alternatives
+
+Tailwind CSS is the recommended utility framework in this proposal, but it is by no means the only utility css framework available. We reviewed and compared a few other alternatives when deciding on which framework to move forward with.
+
+The following list specifies where these alternatives fall short in comparison to Tailwind CSS:
+
+**[Basscss](https://basscss.com/)**
+
+- Less customizable than Tailwind.
```

Comment:
Totally a nice-to-have, but if you remember off-hand per looking at each - it might be helpful to list a few reasons why for each of these are less customizable (this reason occurs in multiple alternatives)

Reply:
I think more-so in comparison to the list of Tailwind's capabilities and customizability the others are less customizable. Tailwind's easy customizability via a JS config object, overriding or extending the library of classes it generates, extensibility via plugins, and ability to still create composed classes all give it a lot of versatility compared to the others which have simpler customization options. I can include the above statement in the description though to make it clearer! Thanks for the suggestion @aaronschachter 😄